### PR TITLE
Register plugins via the app.RegisterPlugins() interface instead of via static initialization

### PR DESCRIPTION
**Pull request by luxas**
ref: #31431

I also came across some packages were the directory name != package name, and made them equal

@smarterclayton @lavalamp @vishh @kubernetes/sig-api-machinery @jbeda @timothysc

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31432)

### CODE REVIEW

**Comment by smarterclayton** [comment]
I strongly approve of this change.  Not sure it's 1.4 though, unless we are really blocked on hyperkube startup time.

File: cmd/hyperkube/kube-apiserver.go [hunk 1]
```diff
@@ -32,6 +32,9 @@ func NewKubeAPIServer() *Server {
 		SimpleUsage: "apiserver",
 		Long:        "The main API entrypoint and interface to the storage system.  The API server is also the focal point for all authorization decisions.",
 		Run: func(_ *Server, args []string) error {
+			// Register all admission and cloudprovider plugins
```
Comment (lavalamp) [inline cmd/hyperkube/kube-apiserver.go]: nit: Can you make the line spacing & comments consistent on these first three files?

**Comment by lavalamp** [comment]
What @smarterclayton said.

**Comment by luxas** [comment]
@lavalamp @smarterclayton Would you like to proceed on this one?

**Comment by smarterclayton** [comment]
Yeah please rebase and I'll give it high priority.

File: plugin/pkg/admission/security/podsecuritypolicy/admission_test.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package admission
+package podsecuritypolicy // import "k8s.io/kubernetes/plugin/pkg/admission/security/podsecuritypolicy"
```
Comment (lavalamp) [inline plugin/pkg/admission/security/podsecuritypolicy/admission_test.go]: Nit: only put the import comment in one file in the package

Comment (luxas) [inline plugin/pkg/admission/security/podsecuritypolicy/admission_test.go]: Okay, will fix that

**Comment by luxas** [comment]
@smarterclayton @lavalamp PTAL, rebased.

Please look at it carefully to make sure I haven't missed some case where we earlier relied on static init.

**Comment by luxas** [comment]
@smarterclayton Updated again in order to fix some lint issues.

I don't have any tools right now to benchmark what the performance gain is, can you test or point me to some tool that can do it?

**Comment by luxas** [comment]
@smarterclayton @lavalamp Is it worth it to have this laying around anymore?
Do you want it or should I close it?

**Pull request closed without merging**